tests/system/src/test/java/test/robot/javafx/scene/ChoiceBoxScrollUpOnCollectionChangeTest.java

```diff
@@ -0,0 +1,181 @@
+ * questions.
+ */
+
+package test.robot.javafx.scene;
+
+import java.util.concurrent.CountDownLatch;
+
+import com.sun.javafx.scene.control.ContextMenuContentShim;
+import com.sun.javafx.tk.Toolkit;
+
+import javafx.application.Application;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.ChoiceBox;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.skin.ChoiceBoxSkin;
+import javafx.scene.control.skin.ChoiceBoxSkinNodesShim;
+import javafx.scene.control.skin.ContextMenuSkin;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.application.Platform;
+import javafx.scene.robot.Robot;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import test.util.Util;
+
+/*
+ * Test for verifying if ChoiceBox items are displayed after
+ * scrolling large number of items and replacing with 2 items.
+ *
+ * There is 1 test in this file.
+ * Steps for testChoicBoxScrollOnCollectionChange()
+ * 1. Create a ChoiceBox and add 50 items to it.
+ * 2. Display the ChoiceBox and scroll down.
+ * 3. Verify if both up and down scroll arrows are displayed.
+ * 4. Replace the ChoiceBox items with 2 items.
+ * 5. Display the ChoiceBox again.
+ * 6. Verify that ChoiceBox items are displayed and no scroll
+ *    arrows are displayed.
+ */
+
+public class ChoiceBoxScrollUpOnCollectionChangeTest {
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static CountDownLatch scrollLatch = new CountDownLatch(1);
+    static CountDownLatch choiceBoxDisplayLatch = new CountDownLatch(1);
+    static CountDownLatch choiceBoxHiddenLatch = new CountDownLatch(1);
+    static Robot robot;
+    static ChoiceBox<String> choiceBox;
+
+    static volatile Stage stage;
+    static volatile Scene scene;
+
+    static final int SCENE_WIDTH = 250;
+    static final int SCENE_HEIGHT = SCENE_WIDTH;
+
+    private void mouseClick(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + x),
+                                (int) (scene.getWindow().getY() + scene.getY() + y));
+            robot.mouseClick(MouseButton.PRIMARY);
+        });
+    }
+
+    private void scrollChoiceBox(int scrollAmt) throws Exception {
+        Util.runAndWait(() -> {
+            for (int i = 0; i < scrollAmt; i++) {
+                robot.keyType(KeyCode.DOWN);
+                Toolkit.getToolkit().firePulse();
+            }
+
+            Util.sleep(400); // Wait for up arrow to get loaded in UI
+
+            robot.keyType(KeyCode.ENTER);
+        });
+    }
+
+    private void showChoiceBox() throws Exception {
+        mouseClick(choiceBox.getLayoutX() + choiceBox.getWidth() / 2,
+                    choiceBox.getLayoutY() + choiceBox.getHeight() / 2);
```

Comment:
are these two lines slightly misaligned?

perhaps we could convert this to
double x = ...;
double y = ...;
mouseClick(x, y);

(there is a value in having one statement per line, in general)

Reply:
> are these two lines slightly misaligned?
> 
The second line is aligned at 6 tabs from the start of the line. Because of that it looks misaligned.

> perhaps we could convert this to double x = ...; double y = ...; mouseClick(x, y);
> 
> (there is a value in having one statement per line, in general)

Since there are no repeated function calls I didn't convert them to variables. Yes converting the long statements to variables would make it more readable. I will work on that.

Reply:
Updated code